src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy

```diff
@@ -236,16 +236,14 @@ class GenerationPlugin implements Plugin<Project> {
 
             if (combined) {
                 // add instrumentation coverage execution data
-                executionData += subProject.fileTree("${subProject.buildDir}/outputs/code_coverage").matching {
-                    include "**/*.ec"
-                }
+
             }
 
             // add if true in extension or for the unit test Jacoco task
             def addToMergeTask = !combined || extension.includeInstrumentationCoverageInMergedReport
 
             if (mergeTask != null && addToMergeTask) {
-                mergeTask.executionData += executionData
```

Comment:
What's the reason behind changing from adding to replacing?

Reply:
Another mistake of mine. I've refactored the whole PR to fix that. The StackOverFlow root cause is the += operator as explained [here](https://github.com/gradle/gradle/issues/8398).